### PR TITLE
feat: no_std eth rpc types

### PR DESCRIPTION
**Pull request by refcell**
### Description

Adds support for `no_std` in `alloy-rpc-types-eth`.

### CODE REVIEW

**Comment by DaniPopes** [comment]
Can you update the workflow and no_std script from `alloy-rs/core`?

**Comment by refcell** [comment]
> Can you update the workflow and no_std script from `alloy-rs/core`? Would like them to be the same

Yes definitely, mind if I do this in another PR to unblock this?

File: crates/rpc-types-eth/src/block.rs [hunk 1]
```diff
@@ -247,13 +249,14 @@ impl HeaderResponse for Header {
 }
 
 /// Error that can occur when converting other types to blocks
-#[derive(Clone, Copy, Debug, thiserror::Error)]
+#[derive(Clone, Copy, Debug)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
```
Comment (mattsse) [inline crates/rpc-types-eth/src/block.rs:253]: the problem with this is that this now removes the display impl for no-std

perhaps we should use derive_more display and manually impl error for all errors @DaniPopes ?

File: crates/rpc-types-eth/src/state.rs [hunk 2]
```diff
@@ -1,9 +1,10 @@
 //! bindings for state overrides in eth_call
 
 use crate::BlockOverrides;
+use alloc::boxed::Box;
 use alloy_primitives::{Address, Bytes, B256, U256};
+use hashbrown::HashMap;
```
Comment (mattsse) [inline crates/rpc-types-eth/src/state.rs:6]: we could either use std hashmap or hashbrown because this change will break in a few places

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
pending @DaniPopes

File: .github/workflows/no_std.yml
Comment (DaniPopes) [inline .github/workflows/no_std.yml:1]: Can you add the job from alloy-rs/core in ci.yml?

File: crates/rpc-types-eth/src/state.rs (hunk 2 above)
Comment (DaniPopes) [reply crates/rpc-types-eth/src/state.rs:6]: this is a breaking change

**Review by DaniPopes — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**